Unit test flake: k8s.io/kubernetes/plugin/pkg/client/auth/oidc

I hit a test flake in `k8s.io/kubernetes/plugin/pkg/client/auth/oidc`

Example build: https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/26807/kubernetes-pull-test-unit-integration/29103/

Failure log:
```
--- FAIL: TestNewOIDCAuthProvider (2.70s)
	oidc_test.go:152: case 1: unexpected error on newOIDCAuthProvider: open /tmp/oidc-cert: no such file or directory
```

Most likely there are two unit tests using this file running in parallel and they raced.

@mml

--- a/plugin/pkg/client/auth/oidc/oidc_test.go
+++ b/plugin/pkg/client/auth/oidc/oidc_test.go
@@ -39,11 +39,16 @@ import (
 )
 
 func TestNewOIDCAuthProvider(t *testing.T) {
-	cert := path.Join(os.TempDir(), "oidc-cert")
-	key := path.Join(os.TempDir(), "oidc-key")
+	tempDir, err := ioutil.TempDir(os.TempDir(), "oidc_test")
+	if err != nil {
+		t.Fatalf("Cannot make temp dir %v", err)
+	}
+	cert := path.Join(tempDir, "oidc-cert")
+	key := path.Join(tempDir, "oidc-key")
 
 	defer os.Remove(cert)
 	defer os.Remove(key)
+	defer os.Remove(tempDir)
 
 	oidctesting.GenerateSelfSignedCert(t, "127.0.0.1", cert, key)
 	op := oidctesting.NewOIDCProvider(t)

